docs/source/guides/speed.rst

```diff
@@ -0,0 +1,279 @@
+       mode='max'
+    )
+    trainer = Trainer(callbacks=[early_stop_callback])
+
+
+Additional parameters that stop training at extreme points:
+
+- ``stopping_threshold``: Stops training immediately once the monitored quantity reaches this threshold.
+  It is useful when we know that going beyond a certain optimal value does not further benefit us.
+- ``divergence_threshold``: Stops training as soon as the monitored quantity becomes worse than this threshold.
+  When reaching a value this bad, we believe the model cannot recover anymore and it is better to stop early and run with different initial conditions.
+- ``check_finite``: When turned on, we stop training if the monitored metric becomes NaN or infinite.
+
+In case you need early stopping in a different part of training, subclass :class:`~pytorch_lightning.callbacks.early_stopping.EarlyStopping`
+and change where it is called:
+
+.. testcode::
+
+    class MyEarlyStopping(EarlyStopping):
+
+        def on_validation_end(self, trainer, pl_module):
+            # override this to disable early stopping at the end of val loop
+            pass
+
+        def on_train_end(self, trainer, pl_module):
+            # instead, do it at the end of training loop
+            self._run_early_stopping_check(trainer, pl_module)
+
+.. note::
+   The :class:`~pytorch_lightning.callbacks.early_stopping.EarlyStopping` callback runs
+   at the end of every validation epoch,
+   which, under the default configuration, happen after every training epoch.
+   However, the frequency of validation can be modified by setting various parameters
+   in the :class:`~pytorch_lightning.trainer.trainer.Trainer`,
+   for example :paramref:`~pytorch_lightning.trainer.trainer.Trainer.check_val_every_n_epoch`
+   and :paramref:`~pytorch_lightning.trainer.trainer.Trainer.val_check_interval`.
+   It must be noted that the `patience` parameter counts the number of
+   validation epochs with no improvement, and not the number of training epochs.
+   Therefore, with parameters `check_val_every_n_epoch=10` and `patience=3`, the trainer
+   will perform at least 40 training epochs before being stopped.
+
+.. seealso::
+    - :class:`~pytorch_lightning.trainer.trainer.Trainer`
+    - :class:`~pytorch_lightning.callbacks.early_stopping.EarlyStopping`
+
+
+Stopping an epoch early
+=======================
+
+You can stop an epoch early by overriding :meth:`~pytorch_lightning.core.hooks.ModelHooks.on_train_batch_start` to return ``-1`` when some condition is met.
+
+If you do this repeatedly, for every epoch you had originally requested, then this will stop your entire run.
+
+----------
+
+.. _amp:
+
+*********************************
+Mixed precision (16-bit) training
+*********************************
+
+Mixed precision is the combined use of both 32 and 16 bit floating points during model training, which reduced memory requirements and improves performance significantly, achiving over 3X speedups on modern GPUs.
```

Comment:
Lets refactor this a little to something like

Mixed precision is the combined use of both 32 and 16 bit floating points to reduce memory footprint during model training, resulting in improved performance, achieving +3X speedups on modern GPUs.